src/index.tsx

```diff
@@ -1093,18 +1098,21 @@ export function callProp(
 }
 
 /**
+ * @deprecated
+ * Prefer using `React.createRef` object-based refs instead of this function.
+ *
  * Get a function to be used with ref={} to register a ref into
- * a variable in the current object
+ * a variable in the current object.
  * @param {Object} elem          React component (usually `this`)
  * @param {String} variableName  Name to save ref (e.g., '_elem' for this._elem)
  * @return {function} handler for ref=}}
  */
-export function registerRef(elem: React.Component<any>, variableName: Keys) {
+export function registerRef<T extends Object>(elem: T, variableName: Keys) {
```

Comment:
You could probably type this `variableName` as `keyof T` instead of `Keys` here

Reply:
Done!